django-stubs/contrib/syndication/views.pyi

```diff
@@ -11,17 +12,160 @@ def add_domain(domain: str, url: str, secure: bool = ...) -> str: ...
 
 class FeedDoesNotExist(ObjectDoesNotExist): ...
 
-class Feed:
-    feed_type: Any = ...
-    title_template: Any = ...
-    description_template: Any = ...
+Item = TypeVar("Item")
+Object = TypeVar("Object")
+
+class Feed(Generic[Item, Object]):
+    feed_type: Type[SyndicationFeed] = ...
+    title_template: Optional[str] = ...
+    description_template: Optional[str] = ...
+    language: Optional[str] = ...
     def __call__(self, request: WSGIRequest, *args: Any, **kwargs: Any) -> HttpResponse: ...
-    def item_title(self, item: Model) -> SafeText: ...
-    def item_description(self, item: Model) -> str: ...
-    def item_link(self, item: Model) -> str: ...
-    def item_enclosures(self, item: Model) -> List[Enclosure]: ...
-    def feed_extra_kwargs(self, obj: None) -> Dict[Any, Any]: ...
-    def item_extra_kwargs(self, item: Model) -> Dict[Any, Any]: ...
-    def get_object(self, request: WSGIRequest, *args: Any, **kwargs: Any) -> Any: ...
+    def get_object(self, request: WSGIRequest, *args: Any, **kwargs: Any) -> Optional[Object]: ...
+    # One of the following three is required
+    def title(self, obj: Object) -> str: ...
+    def title(self) -> str: ...
+    title: str = ...
```

Comment:
These are tricky. The docs says one of them is required/optional, I don't know how to represent that here.

Reply:
Tests fail because of the redefinitions here and below.

Reply:
Maybe `title: Any`?

Reply:
We use this hack when we cannot represent something with types.
Or we can use `title: Union[str, Callable[[Feed], str], Callable[[Feed, Object], str]]`

Reply:
I like the Union approach, I’ll come up with something in the next couple of days 👍🏻

Reply:
How should I write unit tests for this file?